updateAzaanTimers.py

```diff
@@ -5,68 +5,117 @@
-                        help='Volume for azaan (other than fajr) in millibels, 1500 is loud and -30000 is quiet (default 0)')
     return parser
 
-def mergeArgs(args):
-    file_path = pathjoin(root_dir, '.settings')
-    # load values
-    lat = lng = method = fajr_azaan_vol = azaan_vol = None
+def getConfig():
+    # Parse arguments
+    parser = parseArgs()
+    args = parser.parse_args()
+    
+    # Initialise and read config file if present
+    config = ConfigParser()
+    file_path = pathjoin(root_dir, 'settings.ini')
+    config.read(file_path)
+
+    lat = lon = method = fajr_azaan_vol = default_azaan_vol = surahBaqarah = surahVolume = None
+
+    # Get mandatory data. First check args, if not present check settings.ini
     try:
-        with open(file_path, 'rt') as f:
-            lat, lng, method, fajr_azaan_vol, azaan_vol = f.readlines()[0].split(',')
+        if args.lat:
+            lat = float(args.lat)
+            config['DEFAULT']['lat'] = str(lat)
+        else:
+            lat = float(config['DEFAULT']['lat'])
+        
+        if args.lon:
+            lon = float(args.lon)
+            config['DEFAULT']['lon'] = str(lon)
+        else:
+            lon = float(config['DEFAULT']['lon'])
+
+        if args.method:
+            method = args.method
+            config['DEFAULT']['method'] = method
+        else:
+            method = config['DEFAULT']['method']
     except:
-        print('No .settings file found')
-    # merge args
-    if args.lat:
-        lat = args.lat
-    if lat:
-        lat = float(lat)
-    if args.lng:
-        lng = args.lng
-    if lng:
-        lng = float(lng)
-    if args.method:
-        method = args.method
-    if args.fajr_azaan_vol:
-        fajr_azaan_vol = args.fajr_azaan_vol
-    if fajr_azaan_vol:
-        fajr_azaan_vol = int(fajr_azaan_vol)
-    if args.azaan_vol:
-        azaan_vol = args.azaan_vol
-    if azaan_vol:
-        azaan_vol = int(azaan_vol)
-    # save values
-    with open(file_path, 'wt') as f:
-        f.write('{},{},{},{},{}'.format(lat or '', lng or '', method or '',
-                fajr_azaan_vol or 0, azaan_vol or 0))
-    return lat or None, lng or None, method or None, fajr_azaan_vol or 0, azaan_vol or 0 
+        print("Incorrect value or values not provided")
+        lat = lon = method = None
+
+
+    # Get optional data
+    try:
+        if args.default_azaan_vol:
+            default_azaan_vol = int(args.default_azaan_vol)
+        else:
+            default_azaan_vol = int(config['VOLUME']['defaultAzaanVolume'])
+
+        if args.fajr_azaan_vol:
+            fajr_azaan_vol = int(args.fajr_azaan_vol)
+        else:
+            fajr_azaan_vol = int(config['VOLUME']['fajrAzaanVolume'])
+    except:
+        default_azaan_vol = 0
+        fajr_azaan_vol = 0
+
+        
+    config["VOLUME"] = {
+        "defaultAzaanVolume": str(default_azaan_vol), 
+        "fajrAzaanVolume": str(fajr_azaan_vol)
+        }
+
+
+    # Setup Surah Baqarah on Fridays
```

Comment:
This seems super specific IMO. I would like to keep this project simple and only make it an Azaan player. Perhaps start a separate script for surahs? People who wan to play surahs can install and run both scripts. Thoughts?

Reply:
Makes sense.